### PR TITLE
Fix parallel rendering

### PR DESCRIPTION
**Pull request by theKashey**
This PR just provide a proper async flow for parallel rendering (#39) , and meanwhile fixing a strange console-related issue.

### CODE REVIEW

File: src/core/app/Page.ts [hunk 1]
```diff
@@ -29,7 +29,13 @@ export default class Page extends EventEmitter {
     this.options = options;
 
     this.page.on('console', (data: puppeteer.ConsoleMessage) => {
-      consoleHandler(data.type(), data.text());
+      if (typeof data.type === 'function') {
+        consoleHandler(data.type(), data.text());
+      } else {
+        // it IS a string, by fact. Type definitions are wrong
+        // @ts-ignore
+        consoleHandler((data.type as string), (data.text as string));
```
Comment (theKashey) [inline src/core/app/Page.ts]: According to the puppeteer source code (https://github.com/GoogleChrome/puppeteer/blob/a310d57d847b008dba3b0772085840dde3343c64/lib/Page.js#L456) this IS a string.

**Comment by tsuyoshiwada** [comment]
Hi @theKashey, Thank you for PR :heart:

I am sorry..., but I can not confirm PR immediately, so I will confirm later.
Please wait for a little while.

Thanks :pray:

**Comment by theKashey** [comment]
No rush. Thank you for a solution 🔥

**Comment by theKashey** [comment]
Any luck? Both PRs are quite important for this tool, and I am going to open one more.

**Comment by tsuyoshiwada** [comment]
Hi @theKashey, Thank you for contribute!!
Released as `1.1.0-alpha.2` :tada:

**Comment by theKashey** [comment]
🙀💥

**Pull request merged**